### PR TITLE
Add more config parameters to HiGHS python API + set up logging

### PR DESCRIPTION
**Pull request by siwy**
### Reference issue

https://github.com/coin-or/pulp/issues/553  (sort of )

### What does this implement?

Improves the usability of the HiGHS interface:
* Pass more of pulp's parameters directly into HiGHS
* Set-up logging (it's a bit awkward, I wasn't sure if there's a better way to do this)
* Allow callers to specify more options via the options argument

### Any other comments

N/A

### CODE REVIEW

**Comment by siwy** [comment]
Hi @pchtsp - pinging again, would it be possible to merge those changes?

File: pulp/apis/highs_api.py [hunk 1]
```diff
@@ -247,32 +246,56 @@ def actualSolve(self, lp, callback=None):
...
+        #  even with the different logging options set. This callback seems to work, but there
+        #  are probably better ways of doing this ¯\_(ツ)_/¯
+        DEFAULT_CALLBACK = lambda logType, logMsg, callbackValue: print(
+            f"[{logType.name}] {logMsg}"
+        )
+        DEFAULT_CALLBACK_VALUE = ""
+
         def __init__(
             self,
-            mip=True,
```
Comment (pchtsp) [inline pulp/apis/highs_api.py:252]: we prefer having the explicit enumeration of parameters, together with a description. This produces the documentation of the APIs in the docs. It's the only way to know in the docs which parameters are supported and which ones are not.
See https://github.com/coin-or/pulp/blob/a017fdbfa2631dfc31bb8d04c50045e1003c94eb/pulp/apis/cplex_api.py#L296-L316

**Review by pchtsp — COMMENTED** [review]
Thanks for the PR. I added just a minor comment on the arguments, so that it matches the other APIs structure (and also so it shows in the docs).

**Comment by siwy** [comment]
@pchtsp - Thanks for the review, that makes a lot of sense! I've added the explicit parameters and docs back
(sorry for the delay, I had a small bout of flu)

**Comment by pchtsp** [comment]
Hello again and sorry for the delay. In the meantime, we merged some other PRs for highs api. Can you fix the conflicts and re-merge? That way I can approve. thanks again!

**Comment by siwy** [comment]
@pchtsp - all done

**Pull request merged**